leabra7/layer.py

```diff
@@ -68,12 +73,13 @@ def __init__(self, name: str, size: int,
 
         # Desired clamping values
         self.act_ext = torch.Tensor(self.size).zero_()
-        # Last plus phase activation
-        self.acts_p = torch.Tensor(self.size).zero_()
-        # Last minus phase activation
-        self.acts_m = torch.Tensor(self.size).zero_()
-        # Cosine similiarity between acts_p and acts_m, integrated over trials
-        self.cos_diff_avg = 0.0
+        # Phase activation
+        self.phase_acts: Dict[str, torch.Tensor] = dict()
```

Comment:
This stuff (I think) should be in the projection (so we can keep the error-driven learning logic in one place.)